netket/experimental/operator/_fermions_2nd.py

```diff
@@ -217,16 +225,44 @@ def copy(self, *, dtype: Optional[DType] = None):
         op = FermionOperator2nd(
             self.hilbert, [], [], constant=self._constant, dtype=dtype
         )
-        op._operators = copy.deepcopy(self._operators)
+        # careful to make sure we propagate the correct dtype
+        terms = self._operators.keys()
+        weights = np.array(list(self._operators.values()), dtype=dtype)
+        op._operators = dict(zip(terms, weights))
         return op
 
+    def reduce(self):
+        """Reduce the number of operators by bringing them to normal form and grouping equivalent terms"""
+        self._operators = _reduce_operators(self._operators, self.dtype)
+        return self
+
```

Comment:
We should decide if this should be an inplace operation as you made it here (aka, which modifies the existing operator) or if it should be out of place and it does not modify it but returns a new operator which is _reduced_.

In general, out-of-place non modifying APIs are easier to reason and introduce less bugs (especially if we wrap operators inside operators) so I'd lean towards that, but would like to have opinions.

Reply:
yes I agree out of place is easier to debug, however maybe Jannes here means to do a default simplification (i.e. this `reduce` should be called on every instance of the class?) in this case then maybe I would not expose `reduce` as a public function and just make it internal, non returning?